### ISSUE TITLE
[Enhancement] Allow body on REST GET method

### ISSUE THREAD

**Issue by ffazendeiro** [cerberustesting/cerberus-source#2101]
Hi,
We need to send a body message on the GET REST request.
Currently Cerberus is adding the "Service Request" to the URL (the same action as the "Request Detail" does).
Thanks.

**Comment by vertigo17** [comment]
Hello @ffazendeiro ,
what happens is that apache http client 4.5 that we use does not allow to feed entity on GET requests.

For POST:
https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/client/methods/HttpPost.html
We see that setEntity method is available

For GET:
https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/client/methods/HttpGet.html
The setEntity method is not available.

There are lots of discussions on that topic on that link:
https://stackoverflow.com/questions/978061/http-get-with-request-body#:~:text=Yes.,semantic%20meaning%20to%20the%20request.

Basically and in summary, body in GET requests used to be forbidden on the w3c spec of the http protocol. It seems the protocol now allow to define it but those are not advised because most library does not support them and many servers/proxy/router don't even parse it. That seem quite dangerous to use it.

Having said that, it would still be interesting to allow to define it but first step will be to find a JAVA lib that allow that. 

I leave the issue open to process on that analysis.

**Comment by ittaLaRedoute** [comment]
Hi @vertigo17 , any news on that ?
